Update documentation regarding annotations (#123)
laminas/laminas-form: docs/book/form-creation/using-annotations.md

> # Using Annotations or PHP8 Attributes

Reviewer: ```suggestion
# Using PHP8 Attributes or Annotations
```

and fieldsets to the domain model. Wouldn't it be nice to have a central place
to define all of these?

Annotations allow us to solve this problem. With v3, laminas-form supports two

> different types of annotations: PHP8 attributes and traditional PHPDoc annotations.

Reviewer: > PHPDoc

We use "DocBlock" or "docblock" in all documentations, the preferred type is "DocBlock" as is used in [phpDocumentor documentation](https://phpdoc.org).

associated input that has a `StringTrim` filter, and two validators: a
`StringLength` validator indicating the username is between 1 and 25 characters,
and a `Regex` validator asserting it follows a specific accepted pattern. The
form element itself will have an attribute "autofocus", and a label "Username:". 
The "email" element will be of type `Laminas\Form\Element\Email`, and have the label 
"Your email address:".

To use the above, we need `Laminas\Form\Annotation\AttributeBuilder`:

```php
use Laminas\Form\Annotation\AttributeBuilder;

$builder = new AttributeBuilder();
$form    = $builder->createForm(User::class);
```

## Using PHPDoc Annotations

Besides PHP8-only attributes, laminas-form continues to support traditional
PHPDoc annotations with v3, which have been supported with v2 already. Using
these is suitable for legacy projects or if you need support for PHP 7.

To use PHPDoc annotations, include them in your class and/or property docblocks.
Annotation names will be resolved according to the import statements in your
class; as such, you can make them as long or as short as you want depending on
what you import.

> > ### doctrine/annotations dependency

Reviewer: ```suggestion
> ### Installation Requirements
```

>

> > PHPDoc annotations require `doctrine\annotations` as a peer dependency, which 
> > contains an  annotation parsing engine. You need to manually install it using 
> > composer:

Reviewer: ```suggestion
> DocBlock annotations require [Doctrine's annotations parser `doctrine\annotations`](https://www.doctrine-project.org/projects/annotations.html) as a peer dependency, which 
> contains an annotation parsing engine. You need to manually install it using 
> Composer:
```

At this point, you have a form with the appropriate hydrator attached, an input
filter with the appropriate inputs, and all elements.

## List of Supported Annotations

### AllowEmpty

> Marks an input as allowing an empty value. This annotation does
> not require a value. 
> 
> *This annotation is deprecated, please add a `NotEmpty` validator instead.*

Reviewer: ```suggestion
> ### Deprecated
>
> This annotation is deprecated, please add a `NotEmpty` validator instead.

Marks an input as allowing an empty value. This annotation does not require a value. 
```

```php

> /**
> * @AllowEmpty
> */

Reviewer: ```suggestion
/**
 * @AllowEmpty
 */
```
Some more examples contain the same problem.

protected $myProperty;
```

```php
#[AllowEmpty]
protected $myProperty;
```

### Attributes

Used to specify the form, fieldset, or element attributes. This
annotation requires an associative array of values. For PHPDoc annotations,
the array has to be in JSON format.

```php
/**
* @Attributes({"id":"my_property_element","class":"laminas_form"}
*/
protected $myProperty;
```

```php
#[Attributes(["id" => "my_property_element", "class" => "laminas_form"])]
protected $myProperty;
```

### ContinueIfEmpty

> Indicate whether the element can be submitted when it is empty. A boolean
> value is expected. If `@Required` is set to `false`, this needs to be set
> to `true` to allow the field to be empty.
> 
> *This annotation is deprecated, please add a `NotEmpty` validator instead.*

Reviewer: ```suggestion
> ### Deprecated
>
> This annotation is deprecated, please add a `NotEmpty` validator instead.

Indicate whether the element can be submitted when it is empty. A boolean
value is expected. If `@Required` is set to `false`, this needs to be set
to `true` to allow the field to be empty.
```
